src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1145,7 +1141,7 @@ impl Cage {
             if fdguard.is_some() {
                 drop(fdguard);
                 //close the fd in the way of the new fd. If an error is returned from the helper, return the error, else continue to end
-                let close_result = Self::_close_helper_inner(&self, newfd);
+                let _close_result = Self::_close_helper_inner(&self, newfd);
                 // mirror the implementation of linux, ignore the potential error of the close here
             } else { drop(fdguard); }
```

Comment:
Can we change that part like below:

```
 drop(fdguard);
 
 if !fdguard.is_some() {
      let _close_result = Self::_close_helper_inner(&self, newfd);
 }
```

I think this one can be more legible.

Reply:
Your replacement doesn't exactly work because we need to check is_some before the fdguard is dropped. I re-factored a bit to make it cleaner.